shtdlib.sh

```diff
@@ -914,9 +914,11 @@ function on_exit {
         fi
     fi
     debug 10 "Finished cleaning up, de-registering signal trap"
-    # Be a nice Unix citizen and propagate the signal
     trap - EXIT
-    kill -s EXIT ${$}
+    if [ ! $interactive ]; then
```

Comment:
You don't need to put this in the [ ] test command, true/false are commands and can be evaluated directly.